pkg/injection/codemodule/installer/symlink/symlink.go

```diff
@@ -49,6 +50,38 @@ func CreateSymlinkForCurrentVersionIfNotExists(fs afero.Fs, targetDir string) er
 	return nil
 }
 
+func CreateSymlinkForLatestVersion(fs afero.Fs, dk dynakube.DynaKube, latestVersionDir, symlinkDir string) error {
+	// MemMapFs (used for testing) doesn't comply with the Linker interface
+	linker, ok := fs.(afero.Linker)
+	if !ok {
+		log.Info("symlinking not possible", "targetDir", latestVersionDir, "fs", fs)
+
+		return nil
+	}
+
+	log.Info("creating symlink", "points-to(relative)", latestVersionDir, "location", symlinkDir)
+
+	if err := linker.SymlinkIfPossible(latestVersionDir, symlinkDir); err != nil {
+		log.Info("symlinking failed", "codemodules-version", latestVersionDir)
+
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func RemoveSymLink(fs afero.Fs, symLinkPath string) error {
```

Comment:
I think it should be better to use `symlink` instead of `SymLink`, even it's acronym, but after quick look at go source code I see they also follow this standard https://go.dev/src/path/filepath/symlink.go

Reply:
@andriisoldatenko you mean?
```suggestion
func RemoveSymlink(fs afero.Fs, symlinkPath string) error {
```

Reply:
yes, in our code base we use only lowercase, I just checked, so it's important to stay in touch
```
pkg/injection/codemodule/installer/image/installer.go
13:     "github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer/symlink"
88:     if err := symlink.CreateSymlinkForCurrentVersionIfNotExists(installer.fs, targetDir); err != nil {
91:             log.Info("failed to create symlink for agent installation", "err", err)

pkg/injection/codemodule/installer/url/installer.go
13:     "github.com/Dynatrace/dynatrace-operator/pkg/injection/codemodule/installer/symlink"
80:     if err := symlink.CreateSymlinkForCurrentVersionIfNotExists(installer.fs, targetDir); err != nil {
82:             log.Info("failed to create symlink for agent installation", "targetDir", targetDir)

pkg/injection/codemodule/installer/symlink/symlink_test.go
1:package symlink

pkg/injection/codemodule/installer/symlink/config.go
1:package symlink
14:     log = logd.Get().WithName("oneagent-symlink")

pkg/injection/codemodule/installer/symlink/symlink.go
1:package symlink
22:             log.Info("symlinking not possible", "targetDir", targetDir, "fs", fs)
34:     symlinkTargetPath := filepath.Join(targetBindDir, "current")
35:     if fileInfo, _ := fs.Stat(symlinkTargetPath); fileInfo != nil {
36:             log.Info("symlink already exists", "location", symlinkTargetPath)
41:     log.Info("creating symlink", "points-to(relative)", relativeSymlinkPath, "location", symlinkTargetPath)
43:     if err := linker.SymlinkIfPossible(relativeSymlinkPath, symlinkTargetPath); err != nil {
44:             log.Info("symlinking failed", "version", relativeSymlinkPath)
```

Reply:
ok, then your original comment had a typo 😛

Reply:
you mean `Symlink` ? yes the idea is to use as whole word instead of SymLink or symLink etc.

Reply:
I like it 😘  , but the original message confused me a bit 😄

Reply:
it's Friday :) all good

Reply:
will do

Reply:
done https://github.com/Dynatrace/dynatrace-operator/pull/3940/commits/8c1329c701ea50ed9e22fedbe62b8a4c7df182bb